src/main/java/seedu/address/logic/interview/AssignInterviewCommand.java

```diff
@@ -82,6 +84,12 @@ public CommandResult execute(Model model) throws CommandException {
                 throw new CommandException(String.format(MESSAGE_CANDIDATE_DID_NOT_APPLY,
                         candidateIndex.getOneBased(), candidate.getName(), interview.getPositionTitle()));
             }
+
+            if (candidate.hasInterview(assignedInterview)) {
+                throw new CommandException(String.format(MESSAGE_CANDIDATE_HAS_INTERVIEW,
+                        candidateIndex.getOneBased(), candidate.getName(), interview.getDisplayStringWithoutNames()));
+            }
+
```

Comment:
I found a bug. Let's say interview 1 has candidate 1. Then you assign i=1 c=1 2. Error is shown but candidate 2 is assigned. This is the same bug for unassign. Need to have another for loop for candidateIndexes before this for loop to check before executing anything.

Reply:
Actual should be all the checks. Because if you do i=1 c=1 2000, there is a chance that candidate would still be assigned even though an error is thrown for 2000. 
Same for checking for if did not apply. If candidate 5 did not apply for the position but candidate 1 did, assign i=1 c=1 5 may still assign candidate 1 even through error is thrown for candidate 5

Reply:
You can check my PR #258, to see what I meant by separating the checking and the execution phase

Reply:
resolved in latest commit